allow resampling to match model with input

dtreskunov opened the pull request:
Resolves #18 

With this change, the code in #18 which used trigger an exception, now works:
```
+ /opt/python/cp37-cp37m/bin/python repro.py
vosk --min-active=200 --max-active=3000 --beam=10.0 --lattice-beam=2.0 --acoustic-scale=1.0 --frame-subsampling-factor=3 --endpoint.silence-phones=1:2:3:4:5:6:7:8:9:10 --endpoint.rule2.min-trailing-silence=0.5 --endpoint.rule3.min-trailing-silence=1.0 --endpoint.rule4.min-trailing-silence=2.0
LOG (vosk[5.5.599~1-11bed]:ComputeDerivedVars():ivector-extractor.cc:183) Computing derived variables for iVector extractor
LOG (vosk[5.5.599~1-11bed]:ComputeDerivedVars():ivector-extractor.cc:204) Done.
LOG (vosk[5.5.599~1-11bed]:RemoveOrphanNodes():nnet-nnet.cc:948) Removed 1 orphan nodes.
LOG (vosk[5.5.599~1-11bed]:RemoveOrphanComponents():nnet-nnet.cc:847) Removing 2 orphan components.
LOG (vosk[5.5.599~1-11bed]:Collapse():nnet-utils.cc:1472) Added 1 components, removed 2
LOG (vosk[5.5.599~1-11bed]:CompileLooped():nnet-compile-looped.cc:345) Spent 0.0140388 seconds in looped compilation.
False
OK
```

Commit:
allow resampling to match model with input

dtreskunov commented:
Note: this is also achievable by adding the following lines into the model's `mfcc.conf`:

```
--allow-upsample=true
--allow-downsample=true
```

nshmyrev commented:
Sorry, this would lead to accuracy issues which will be hidden from the user. You can downsample, but you can not upsample the audio, the accuracy will drop. It is not possible to decode 8khz audio with 16khz model without significant drop in the accuracy unless the model is trained in a special way with double sample rate.

I'm ok to leave just allow_downsample so that higher sampling rates will work.

Commit:
add methods to get model freq and allow up/down sampling

update example code to use new methods

dtreskunov commented:
Please take a look at the latest changes. I added three new methods:

1. `Model::GetSampleFrequency` returns the sample rate that the model was trained on
1. `Model::SetAllowDownsample` enables downsampling
1. `Model::SetAllowUpsample` enables upsampling

I also updated the test script to use these new methods. It uses the actual sample rate from the WAV file to initialize the recognizer, and enables up- or down-sampling in case the model doesn't match.

dtreskunov committed:
change float GetSampleFrequency to int SampleFrequency

dtreskunov commented:
Hi, can you please review and merge/reject this PR?

nshmyrev commented:
Thank you, I've just pushed https://github.com/alphacep/vosk-api/commit/378fa3499abc804a3b47d4785fa09550aa995f39 which is a simplified version of this. It should be good now.

dtreskunov commented:
Hi, sorry to bother you (again!) but I think it would be useful to expose a method for getting the model frequency. It would be helpful to capture microphone audio at the correct rate without needing to downsample needlessly.
